scr/hw12/Box.java

```diff
@@ -0,0 +1,46 @@
+package hw12;
+
+public class Box <T extends Fruit> {
+    private int amount;
+    private T kind;
+
+    public Box(T kind){
+        this.kind = kind;
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+
+    public void addOne() {
+        setAmount(getAmount() + 1);
+    }
+
+    public void addMany(int amount) {
+        setAmount(getAmount() + amount);
+    }
+
+    public float getWeight() {
+        return getAmount() * kind.getWeight();
+    }
+
+    public boolean compare(Box box){
+        if ((getWeight() != box.getWeight())) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean merge(Box box){
```

Comment:
Box must be parameterized to avoid using getClass()
`public boolean merge(Box<T> box){...}`